brave/brave-core
# Handle microsoft-edge: protocol on Windows

When Brave is set as a default app for microsoft-edge protocol,
Windows will pass "microsoft-edge:XXX" as command line arguments.
This edge protocol includes url information.
To make browser load passed url included in edge protocol, browser
should parse ms-edge protocol args fetch url from it.
Then, new browser or existing browser window will load that url.


Resolves https://github.com/brave/brave-browser/issues/13875

Security review: https://github.com/brave/security/issues/552

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
`npm run test brave_unit_tests -- --filter=*MSEdgeProtocolTest*`

------

Use the Windows `Defaults` program to set the `microsoft-edge` handler manually
![image](https://user-images.githubusercontent.com/4733304/130571703-78c0b62b-9b6f-4c40-94a3-d16107785ccd.png)

------

### Test 1 -  from chrome uninstall process
1. Set brave as a default app for microsoft-edge protocol
2. Install chrome and then uninstall
3. Check uninstall survey page is loaded in brave

### Test 2 - search from windows task bar
1. Set brave as a default app for microsoft-edge protocol
2. Search any keyword from windows taskbar.
3. Check bing search url is loaded in brave
4. Search another keyword from windows taskbar
5. Check another tab is created and new tab loads bing searcu url

### Test 3 - uninstall
1. Set brave as a default app for microsoft-edge protocol
2. Uninstall brave
3. Search any keyword from windows taskbar
4. Check Windows asks users to set default app for microsoft-edge protocol again
